packages/rpc-core/src/rpc-methods/__tests__/get-block-commitment-test.ts

```diff
@@ -0,0 +1,36 @@
+import { createHttpTransport, createJsonRpc } from '@solana/rpc-transport';
+import type { Rpc } from '@solana/rpc-transport/dist/types/json-rpc-types';
+import fetchMock from 'jest-fetch-mock-fork';
+
+import { createSolanaRpcApi, SolanaRpcMethods } from '../index';
+
+describe('getBlockCommitment', () => {
+    let rpc: Rpc<SolanaRpcMethods>;
+    beforeEach(() => {
+        fetchMock.resetMocks();
+        fetchMock.dontMock();
+        rpc = createJsonRpc<SolanaRpcMethods>({
+            api: createSolanaRpcApi(),
+            transport: createHttpTransport({ url: 'http://127.0.0.1:8899' }),
+        });
+    });
+
+    it('returns the block commitment for an older block, which has null commitment', async () => {
+        expect.assertions(1);
+        const getBlockCommitmentPromise = rpc.getBlockCommitment(BigInt(0)).send();
+        await expect(getBlockCommitmentPromise).resolves.toMatchObject({
+            commitment: null,
+            totalStake: expect.any(BigInt),
+        });
+    });
+
+    it('returns the block commitment for a recent block', async () => {
+        expect.assertions(1);
+        const slot = await rpc.getSlot({ minContextSlot: 0n }).send();
```

Comment:
Tests like this worry me because:

1. Now we're testing _two_ RPC methods in one test
2. If `getSlot` ever returned `0n`, then this test would fail

I would rather have _no_ test than a test that could fail transiently.

Until we have [this level of control](https://github.com/solana-labs/solana/blob/3cc97c75e67b710124cf5749476bef92f7004de8/core/src/replay_stage.rs#L4727-L4748) over the test validator, I think it might be better to just note the cases that we wish we could tee up consistently, but can't with `it.todo`.

```ts
it.todo('...')
```

Reply:
Ooooh yeah that's nifty. Do we have some kind of roadmap to this level of control with JS?

Reply:
I think we either need to break out our 🦀 and make some changes to the test validator itself, or we need to find a champion in core eng.

It's just never as important as other things.